torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4189,6 +4211,114 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            sample_inputs_func=sample_inputs_polar),
+    # To test reference numerics against multiple values of argument `n`,
+    # we make multiple OpInfo entries with each entry corresponding to different value of n (currently 0 to 4).
+    # We run the op tests from test_ops.py only for `n=0` to avoid redundancy in testing.
+    UnaryUfuncInfo('polygamma',
+                   op=lambda x, n, **kwargs: torch.polygamma(n, x, **kwargs),
+                   variant_test_name='polygamma_n_0',
+                   ref=reference_polygamma if TEST_SCIPY else _NOTHING,
+                   dtypes=floating_types(),
+                   dtypesIfCPU=floating_types(),
+                   dtypesIfCUDA=floating_types_and(torch.half),
+                   sample_inputs_func=sample_inputs_polygamma,
+                   skips=(
+                       # Probably related to the way the function is
+                       # scripted for JIT tests (or maybe not).
+                       # RuntimeError:
+                       # Arguments for call are not valid.
+                       # The following variants are available:
+                       #   aten::polygamma(int n, Tensor self) -> (Tensor):
+                       #   Expected a value of type 'Tensor' for argument 'self' but instead found type 'int'.
+                       #   aten::polygamma.out(int n, Tensor self, *, Tensor(a!) out) -> (Tensor(a!)):
+                       #   Expected a value of type 'Tensor' for argument 'self' but instead found type 'int'.
+                       # The original call is:
+                       #   File "<string>", line 3
+                       # def the_method(i0):
+                       #     return torch.polygamma(i0, 1)
+                       #            ~~~~~~~~~~~~~~~ <--- HERE
+                       SkipInfo('TestCommon', 'test_variant_consistency_jit'),),
+                   sample_kwargs=lambda device, dtype, input: ({'n': 0}, {'n': 0})),
+    UnaryUfuncInfo('polygamma',
+                   op=lambda x, n, **kwargs: torch.polygamma(n, x, **kwargs),
+                   variant_test_name='polygamma_n_1',
+                   ref=reference_polygamma if TEST_SCIPY else _NOTHING,
+                   dtypes=floating_types(),
+                   dtypesIfCPU=floating_types(),
+                   dtypesIfCUDA=floating_types_and(torch.half),
+                   sample_inputs_func=sample_inputs_polygamma,
+                   skips=(
+                       # Redundant tests
+                       SkipInfo('TestGradients'),
+                       SkipInfo('TestOpInfo'),
+                       SkipInfo('TestCommon'),
+                       # Mismatch: https://github.com/pytorch/pytorch/issues/55357
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_extremal'),
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_hard'),
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_normal'),
```

Comment:
I'm surprised we have to skip normal here, too. From the issue I thought only extremal values were divergent?

Reply:
This scenario is mentioned in the issue https://github.com/pytorch/pytorch/issues/55357 under for `n=1`.

Reply:
Also note that.
`n=1` has a special path (different from n > 1)
https://github.com/pytorch/pytorch/blob/2219286de4e372371198bc85e15872093fb6fb0e/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L389-L400

Reply:
Aha, thank you for correcting me. You're right, of course.

This behavior only happens near the poles, right? What if restrict the domain of the function to work around this issue for now? That would expand test coverage.

Reply:
Unfortunately, `reference_tests` don't actually allow for specifying domain.

https://github.com/pytorch/pytorch/blob/2219286de4e372371198bc85e15872093fb6fb0e/test/test_unary_ufuncs.py#L116-L143
The note mentions that `The randomly generated values can be restricted by the domain  argument.` however not the final tensors to be tested

Here the vals are passed on without filtering the values based on domain.
https://github.com/pytorch/pytorch/blob/2219286de4e372371198bc85e15872093fb6fb0e/test/test_unary_ufuncs.py#L79-L83

Reply:
Darn; good point.

I don' want to expand the scope of this PR but I'd like to consider follow-up PRs for:

- fixing polygamma
- filtering vals here 
 
so that we can enable this test for polygamma with n=1.

Reply:
Noted!